consensus/reactor.go

```diff
@@ -272,21 +272,57 @@ func (conR *Reactor) ReceiveEnvelope(e p2p.Envelope) {
 			conR.conS.mtx.Lock()
 			initialHeight := conR.conS.state.InitialHeight
 			conR.conS.mtx.Unlock()
+			schema.WriteConsensusState(
+				conR.traceClient,
+				msg.Height,
+				msg.Round,
+				uint8(msg.Step),
+				string(e.Src.ID()),
+				schema.ConsensusNewRoundStep,
+				schema.Download,
+			)
 			if err = msg.ValidateHeight(initialHeight); err != nil {
 				conR.Logger.Error("Peer sent us invalid msg", "peer", e.Src, "msg", msg, "err", err)
 				conR.Switch.StopPeerForError(e.Src, err)
 				return
 			}
 			ps.ApplyNewRoundStepMessage(msg)
 		case *NewValidBlockMessage:
+			schema.WriteConsensusState(
+				conR.traceClient,
+				msg.Height,
+				msg.Round,
+				uint8(ps.PRS.Step),
+				string(e.Src.ID()),
+				schema.ConsensusNewValidBlock,
+				schema.Download,
+			)
 			ps.ApplyNewValidBlockMessage(msg)
 		case *HasVoteMessage:
 			ps.ApplyHasVoteMessage(msg)
+			schema.WriteConsensusState(
+				conR.traceClient,
+				msg.Height,
+				msg.Round,
+				uint8(msg.Type), // NOTE: msg.Type is not the sending node's step
+				string(e.Src.ID()),
+				schema.ConsensusHasVote,
+				schema.Download,
+			)
 		case *VoteSetMaj23Message:
 			cs := conR.conS
 			cs.mtx.Lock()
 			height, votes := cs.Height, cs.Votes
 			cs.mtx.Unlock()
+			schema.WriteConsensusState(
+				conR.traceClient,
+				msg.Height,
+				msg.Round,
+				uint8(msg.Type),
+				string(e.Src.ID()), // NOTE: msg.Type is not the sending node's step
```

Comment:
not sure what this means but should it apply to the line above?

```suggestion
				uint8(msg.Type), // NOTE: msg.Type is not the sending node's step
				string(e.Src.ID()),
```

Reply:
thanks good catch